src/ContactPage/Contact.js

```diff
@@ -0,0 +1,125 @@
+import React from "react";
+import "./Contact.scss";
+
+class Contact extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      name: "",
+      email: "",
+      subject: "",
+      message: "",
+      errors: {
+        name: "",
+        email: "",
+        subject: "",
+        message: ""
+      }
+    };
+  }
+
+  handleChange = event => {
+    event.preventDefault();
+    const validEmailRegex = RegExp(
+      /^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i
+    );
+
+    const { name, value } = event.target;
+    let errors = this.state.errors;
+
+    switch (name) {
+      case "name":
+        errors.name = value.length < 5 ? "Please enter your full name!" : "";
+        break;
+      case "email":
+        errors.email = validEmailRegex.test(value) ? "" : "Email is not valid!";
+        break;
+      case "subject":
+        errors.subject = value.length < 3 ? "Please enter a subject!" : "";
+        break;
+      case "message":
+        errors.message = value.length < 5 ? "Please enter a message" : "";
+      default:
+        break;
+    }
+
+    this.setState({ errors, [name]: value });
+  };
+
+  canBeSubmitted = () => {
+    const filledIn =
+      this.state.name.length > 0 &&
+      this.state.email.length > 0 &&
+      this.state.subject.length > 0 &&
+      this.state.message.length > 0;
+    const { name, email, subject, message } = this.state.errors;
```

Comment:
Good practice to put all your destructuring and variable initialisers at the top of the function